rpcs3/Emu/Cell/Modules/cellPad.cpp

```diff
@@ -84,38 +84,38 @@ void cellPad_NotifyStateChange(u32 index, u32 state)
 
 	std::lock_guard lock(pad::g_pad_mutex);
 
-	if (!info->max_connect)
+	if (index >= info->get_max_connect())
 	{
 		return;
 	}
 
-	const auto handler = pad::get_current_handler();
-
-	const auto& pads = handler->GetPads();
-
-	const u32 old = info->reported_info[index].port_status;
+	pad_data_internal& reported_info = info->reported_info[index];
+	
+	const u32 oldStatus = reported_info.port_status;
 
-	// Ignore sent status for now, use the latest instead
```

Comment:
I think using the latest status instead of sent one with the event is fine, this event system seems meant to be a notification for the thread to update state.

Reply:
Updated to always use the entire status and added comments.